src/pebble.py

```diff
@@ -69,6 +71,31 @@ def replan_mjolnir(container: ops.model.Container) -> None:
     container.replan()
 
 
+def replan_stats_exporter(container: ops.model.Container, charm_state: CharmState) -> None:
+    """Replan Synapse StatsExporter service.
+
+    Args:
+        container: Charm container.
+        charm_state: Instance of CharmState.
+    """
+    layer = _stats_exporter_pebble_layer()
+    datasource = charm_state.datasource
+    if datasource is not None:
+        layer["services"][STATS_EXPORTER_SERVICE_NAME]["environment"] = {
+            "PROM_SYNAPSE_DATABASE": datasource["db"],
+            "PROM_SYNAPSE_HOST": datasource["host"],
+            "PROM_SYNAPSE_PORT": datasource["port"],
+            "PROM_SYNAPSE_USER": datasource["user"],
+            "PROM_SYNAPSE_PASSWORD": datasource["password"],
+        }
+        try:
+            container.add_layer(STATS_EXPORTER_SERVICE_NAME, layer, combine=True)
+            container.start(STATS_EXPORTER_SERVICE_NAME)
+        except ops.pebble.Error as e:
+            logger.debug("Ignoring error while restarting Synapse Stats Exporter")
```

Comment:
What is the reason for ignoring please?

Reply:
Ok, do you think that is worth adding as a comment? Ignoring errors is unusual so anyone reading the code might have a similar thought 🙂

Reply:
Sure, added!